### PR TITLE
Use asyncio get_event_loop to get loop

### PR DESCRIPTION
**Pull request by r-f-g**
- remove usage of loop from juju

fixes: https://bugs.launchpad.net/juju-verify/+bug/1954308

### CODE REVIEW

File: juju_verify/verifiers/base.py [hunk 1]
```diff
@@ -198,26 +197,14 @@ def check_has_sub_machines(self) -> Result:
...
-        task_map = {
-            child_tag: parent_child_pair.child.is_leader_from_status()
-            for child_tag, parent_child_pair in parent_child_pairs.items()
-        }
-
-        loop = asyncio.get_event_loop()
-
-        results = loop.run_until_complete(asyncio.gather(*task_map.values()))
-        result_map = dict(zip(task_map.keys(), results))
-
```
Comment (mkalcok) [inline juju_verify/verifiers/base.py:210]: This change seems unrelated to the removal of the event loop. What's the story here?

Comment (r-f-g) [reply juju_verify/verifiers/base.py:210]: Ah sorry, you are right. This change should not be here.

This is used to get knowledge if unit is leader and that use it to provide `*` in unit name.
e.g.  ```bash
# with information about leader
[WARN] ceph-osd/0 has units running on child machines: ceph-mon/0*
# without the information
[WARN] ceph-osd/0 has units running on child machines: ceph-mon/0
```
IMO:  This is just waste of time to collect this information. It's not really useful.

Comment (mkalcok) [reply juju_verify/verifiers/base.py:210]: I also don't quite understand the need to check for unit leader. Maybe @xavpaice could provide some insight on this as he designed this check?

Comment (r-f-g) [reply juju_verify/verifiers/base.py]: I have reversed these changes and we can redo it in another PR.

File: tests/functional/tests/base.py [hunk 2]
```diff
@@ -15,6 +15,8 @@
 class BaseTestCase(unittest.TestCase):
     """Base class for functional testing of verifiers."""
 
+    loop = asyncio.get_event_loop()
```
Comment (mkalcok) [inline tests/functional/tests/base.py:18]: Is this 100% safe approach? the `asyncio.get_event_loo()` gets executed at the moment when this module is first imported (and never again). Will this class attribute have correct reference to the current event loop?
Maybe it'd be better to implement it as a caching property?

**Review by mkalcok — DISMISSED** [review]

**Review by mkalcok — APPROVED** [review]

**Review by r-f-g — COMMENTED** [review]
recheck-full

**Review by r-f-g — COMMENTED** [review]
recheck-full

**Review by mkalcok — APPROVED** [review]
